packages/flame/lib/src/collisions/collision_callbacks.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:meta/meta.dart';
+
+import '../../components.dart';
+
+/// The [CollidableType] is used to determine which other collidables that it
+/// should collide with.
+enum CollidableType {
+  /// Collides with other collidables of type active or passive.
+  active,
+
+  /// Collides with other collidables of type active.
+  passive,
+
+  /// Will not collide with any other collidables.
+  inactive,
+}
+
+/// The [GenericCollisionCallbacks] mixin can be used to get callbacks from the
+/// collision detection system, potentially without using the Flame component
+/// system.
+/// The default implementation used with FCS is [CollisionCallbacks].
+mixin GenericCollisionCallbacks<T> {
+  Set<T>? _activeCollisions;
+
+  /// The objects that the object is currently colliding with.
+  Set<T> get activeCollisions => _activeCollisions ??= {};
+
+  /// Whether the object is currently colliding or not.
+  bool get isColliding {
+    return _activeCollisions != null && _activeCollisions!.isNotEmpty;
+  }
+
+  /// Whether the object is colliding with [other] or not.
+  bool collidingWith(T other) {
+    return _activeCollisions != null && activeCollisions.contains(other);
+  }
+
+  /// [onCollision] is called in every tick when this object is colliding with
+  /// [other].
+  @mustCallSuper
+  void onCollision(Set<Vector2> intersectionPoints, T other) {
+    onCollisionCallback?.call(intersectionPoints, other);
+  }
+
+  /// [onCollisionStart] is called in the first tick when this object starts
+  /// colliding with [other].
+  @mustCallSuper
+  void onCollisionStart(Set<Vector2> intersectionPoints, T other) {
+    activeCollisions.add(other);
+    onCollisionStartCallback?.call(intersectionPoints, other);
+  }
+
+  /// [onCollisionEnd] is called once when this object has stopped colliding
+  /// with [other].
+  @mustCallSuper
+  void onCollisionEnd(T other) {
+    activeCollisions.remove(other);
+    onCollisionEndCallback?.call(other);
+  }
+
+  /// Assign your own [CollisionCallback] if you want a callback when this
+  /// shape collides with another [T].
+  CollisionCallback<T>? onCollisionCallback;
+
+  /// Assign your own [CollisionCallback] if you want a callback when this
+  /// shape starts to collide with another [T].
+  CollisionCallback<T>? onCollisionStartCallback;
+
+  /// Assign your own [CollisionEndCallback] if you want a callback when this
+  /// shape stops colliding with another [T].
+  CollisionEndCallback<T>? onCollisionEndCallback;
+}
+
+mixin CollisionCallbacks on Component
+    implements GenericCollisionCallbacks<PositionComponent> {
+  @override
+  Set<PositionComponent>? _activeCollisions;
+  @override
+  Set<PositionComponent> get activeCollisions => _activeCollisions ??= {};
+
+  @override
+  bool get isColliding {
+    return _activeCollisions != null && _activeCollisions!.isNotEmpty;
+  }
```

Comment:
This seems to be exactly same as in the parent class.

Reply:
Yes, but since mixins can't extend we'll have to re-implement it here.